svsim/src/main/resources/simulation-driver.cpp

```diff
@@ -914,6 +914,10 @@ void simulation_main(int argc, char const **argv) {
 }
 
 void run_simulation(int delay) {
+  if(!delay) {
+    testbench->eval_step();
+    return;
+  }
   testbench->eval();
   context->timeInc(delay);
```

Comment:
```suggestion
  testbench->eval();
  if (delay) {
    context->timeInc(delay);
  }
```
I think this would be a little cleaner

Reply:
This will still have extra waveform dump behavior:
<img width="1379" alt="图片" src="https://github.com/user-attachments/assets/1e7ef59b-5735-4f0c-a69e-d061ce1f86cc" />
I think this problem is caused by the difference between `eval()` and `eval_step()`, see: https://github.com/verilator/verilator/issues/2832#issuecomment-797150225 and https://verilator.org/guide/latest/connecting.html#wrappers-and-model-evaluation-loop

Reply:
I see, thanks for digging into this. Those links seem to suggest that if you call `eval_step()` you also must call `eval_end_step()`, do we need to do that here?

Reply:
I think we don't need to call `eval_end_step()` here.
The `eval()` function wrap `eval_step()` and `eval_end_step()` together, and the `eval_end_step()` will trigger tracing. If the require statement in ChiselSim/svsim expect to update the design model without waveform dump, then we just call `eval_step()` here, and the `eval_end_step()` will be called in the next step ( wrapped in `eval()`).